src/factory/mod.rs

```diff
@@ -468,7 +485,44 @@ impl Factory {
         Source::with_object(object)
     }
 
-    /// Update the geometry of `DynamicMesh`.
+    /// Update the positions of the geometry of a `DynamicMesh`.
+    pub fn overwrite_position(
+        &mut self,
+        mesh: &mut DynamicMesh,
+        index: usize,
+        position: mint::Point3<f32>,
+    ) {
+        self.hub.lock().unwrap().update_mesh(mesh);
+
+        let normal = if let Some(n) = mesh.geometry.base_shape.normals.get(index) {
+            [f2i(n.x), f2i(n.y), f2i(n.z), I8Norm(0)]
+        } else {
+            NORMAL_Z
+        };
+
+        let tangent = if let Some(t) = mesh.geometry.base_shape.tangents.get(index) {
+            [f2i(t.x), f2i(t.y), f2i(t.z), f2i(t.w)]
+        } else {
+            TANGENT_X
+        };
+
+        let uv = if let Some(uv) = mesh.geometry.base_shape.tex_coords.get(index) {
+            uv.clone().into()
+        } else {
+            [0.0, 0.0]
+        };
+
+        mesh.geometry.base_shape.vertices[index] = position;
```

Comment:
index can be invalid, need to check

Reply:
What would we do if the index is out of range?